frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelExclusionsFilter.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { Button, Col, Row, Select } from 'antd'
+import { useActions, useValues } from 'kea'
+import equal from 'fast-deep-equal'
+import useSize from '@react-hook/size'
+import { DeleteOutlined } from '@ant-design/icons'
+import { ActionFilter } from 'scenes/insights/ActionFilter/ActionFilter'
+import { funnelLogic } from 'scenes/funnels/funnelLogic'
+import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { ANTD_TOOLTIP_PLACEMENTS, areObjectValuesEmpty, clamp } from 'lib/utils'
+import { FunnelExclusionEntityFilter, ActionFilter as ActionFilterType, EntityTypes } from '~/types'
+
+function ExclusionRowSuffix({
+    filter,
+    index,
+    onClose,
+    isVertical,
+}: {
+    filter: ActionFilterType | FunnelExclusionEntityFilter
+    index: number
+    onClose?: () => void
+    isVertical: boolean
+}): JSX.Element | null {
+    const { exclusionFilters, exclusionDefaultStepRange, areFiltersValid, numberOfSeries } = useValues(funnelLogic)
+    const { setOneEventExclusionFilter } = useActions(funnelLogic)
+
+    const persistedStepRange = {
+        funnel_from_step: exclusionFilters.events?.[index]?.funnel_from_step,
+        funnel_to_step: exclusionFilters.events?.[index]?.funnel_to_step,
+    }
+
+    const [localStepRange, setLocalStepRange] = useState<Omit<FunnelExclusionEntityFilter, 'id' | 'name'>>(
+        !areObjectValuesEmpty(persistedStepRange) ? persistedStepRange : exclusionDefaultStepRange
+    )
+
+    const setExclusionRowValue = (): void => {
+        setOneEventExclusionFilter({ ...filter, ...localStepRange }, index)
+    }
+
+    const onBlur = (): void => {
+        if (!equal(persistedStepRange, localStepRange)) {
+            setExclusionRowValue()
+        }
+    }
+
+    const onChange = (fromStep: number, toStep: number): void => {
+        const funnel_from_step = clamp(fromStep, 0, exclusionDefaultStepRange.funnel_to_step - 1)
+        setLocalStepRange({
+            funnel_from_step,
+            funnel_to_step: clamp(toStep, funnel_from_step + 1, exclusionDefaultStepRange.funnel_to_step),
+        })
+    }
+
+    useEffect(() => {
+        onChange(localStepRange.funnel_from_step, localStepRange.funnel_to_step)
+    }, [exclusionDefaultStepRange])
```

Comment:
Is there some way we can avoid the local state here? Introducing local state always leads to some way things can get out of sync. If there's a way to directly derive this state what what we have in a logic, the widget will be more durable.

Reply:
Yup we can avoid this. Just made some changes getting rid of this local state stuff.